### ISSUE TITLE
0.31 Filters no longer render on questions using a saved SQL question as a data sources

### ISSUE THREAD

**Issue by nickhamlin** [metabase/metabase#8971]
Your browser and the version: Chrome 70.0.3538.67 (Official Build) (64-bit)
Your operating system: OS X 10.12.6
Your databases: MySQL
Metabase version: 0.31.0
Metabase hosting environment: Docker
Metabase internal database: MySQL

Steps to produce
1. Save a question using the SQL editor. I used the sample dataset with the query `select * from ORDERS where PRODUCT_ID=14`
2. Make a new question in the query builder with your new saved question as a data source
3. Add a filter. I used `User is greater than 100`
4. Save this new question. Upon saving, the filter will disappear, though an X will still be visible.

I can see some JS errors in the console when I do this
```
app-main.bundle.js?78e15b587bf48274efb7:5 Error: getFieldRefFromColumn expects non-null fieldId or column with non-null id
    at t.getFieldRefFromColumn (app-main.bundle.js?78e15b587bf48274efb7:5)
    at t.default (app-main.bundle.js?78e15b587bf48274efb7:5)
    at app-main.bundle.js?78e15b587bf48274efb7:5
    at Array.map (<anonymous>)
    at e.value (app-main.bundle.js?78e15b587bf48274efb7:5)
    at t.value (app-main.bundle.js?78e15b587bf48274efb7:5)
    at n.visualizationIsClickable (app-main.bundle.js?78e15b587bf48274efb7:5)
    at t.value (app-main.bundle.js?78e15b587bf48274efb7:5)
    at t.n.tableHeaderRenderer (app-main.bundle.js?78e15b587bf48274efb7:5)
    at app-main.bundle.js?78e15b587bf48274efb7:5
```

I'm wondering if this might be related to https://github.com/metabase/metabase/issues/8940, since it seems like another situation where field metadata isn't getting loaded properly? Please let me know if there's more info I can provide!

**Comment by nickhamlin** [comment]
Just got another report from one of our users that the "click to drilldown" functionality that used to exist on these questions is also now gone. I'm wondering if it might be related to this underlying issue as well?

**Comment by RMMSecurity** [comment]
> Just got another report from one of our users that the "click to drilldown" functionality that used to exist on these questions is also now gone. I'm wondering if it might be related to this underlying issue as well?

#8854 #8922 

It looks like custom expression portion was patched, but not the saved questions (Nested queries) it's missing the column.id when grouping on a saved question. The dataset api call doesn't return and id. I can get it to work in dev instances only by specifying the column.id so it'll probably have to be passed when fetching the dataset for aggregates of nested queries.

**Comment by DonRicardoViteri** [comment]
Hello we are discussing this at http://discourse.metabase.com/t/filters-are-not-showing/4971/11?u=rviteri

Basically there is an issue with the filters not showing when there is a question from a native query.  

![image](https://user-images.githubusercontent.com/34017819/51416304-dd8a4d80-1b46-11e9-98f4-f527f7571e2b.png)

**Comment by flamber** [comment]
If you create a nested question and add filters and run it, then it still shows filters and it makes a request to `/api/dataset`, where the `query.filter` response is:
```JSON
"filter":["time-interval",["field-id",["field-literal","CREATED_AT","type/DateTime"]],-30,"day",{}]
```
When saving the nested question (viewing an existing nested question), then it makes a request to `/api/card` (same for `/api/query`), but filters are mangled and the `query.filter` response is:
```JSON
"filter":["time-interval",["field-literal","CREATED_AT","type/DateTime"],-30,"day",{}]
```
Been digging around and seems like it's related to this:
https://github.com/metabase/metabase/blob/5db449c6e153fdc4144652b2f3ac6caa413b845d/src/metabase/mbql/util.clj#L280-L296
The older versions (tested with 0.30.0), which used to work, had this little warning:
```
WARN middleware.expand :: It doesn't make sense to use `field-literal` forms inside `field-id` forms.
Instead of [field-id [field-literal ...]], just do [field-literal ...].
```
So now the question is: Is this a backend regression or does the frontend need to adjust for this?
I've been trying to mess around in the frontend, but without much success. Backend, I'm lost :-)

**Comment by codenamesubho** [comment]
Facing this issue as of metabase v0.32.4.

**Comment by mazameli** [comment]
Bumping this up to a P1. It kind of breaks a key use case with nested queries, and it seems a fair number of folks are running into this.

**Comment by almeidajeff** [comment]
Hi guys,

I'm facing this same problem in this latest 0.32.5 release, some palatative solution?

I was able to identify the values of the filters through json's response from the browser, but that does not help much :(

**Comment by mazameli** [comment]
This is fixed in `release-0.33.x`, i.e. in the [release candidate for 0.33](https://github.com/metabase/metabase/releases/tag/v0.33.0-RC1).  🎉 

![image](https://user-images.githubusercontent.com/2223916/62327551-cae1c100-b465-11e9-8e3a-fe36fd46e6e0.png)
